src/boat_simulator/boat_simulator/nodes/physics_engine/physics_engine_node.py

```diff
@@ -310,13 +317,18 @@ def __init_timer_callbacks(self):
     # PUBLISHER CALLBACKS
     def __publish(self):
         """Synchronously publishes data to all publishers at once."""
+        self.__update_sim_wind_sensor()
         self.__update_boat_state()
         # TODO Get updated boat state and publish (should this be separate from publishing?)
         # TODO Get wind sensor data and publish (should this be separate from publishing?)
         self.__publish_gps()
         self.__publish_wind_sensors()
         self.__publish_kinematics()
         self.__publish_counter += 1
+        
+    def __update_sim_wind_sensor(self):
+        """Updates wind parameter of wind sensor"""
+        self.update(wind=self.__wind_generator.next())
```

Comment:
I believe you could use the public setter provided for this attribute. So you could replace the following code with `self.__sim_wind_sensor = self.__wind_generator.next()`. See https://github.com/UBCSailbot/sailbot_workspace/blob/3bf448c71e83f4cc4cbba4d6527c837414a02b9c/src/boat_simulator/boat_simulator/common/sensors.py#L126 in the `SimWindSensor` class. See this [link](https://www.freecodecamp.org/news/python-property-decorator/) for more details on python getters and setters.